### PR TITLE
Hide activate and customize options for WP.org themes on non-atomic sites

### PR DESCRIPTION
**Pull request by WBerredo**
#### Proposed Changes

Hide activate and customize options for WP.org themes on non-atomic sites

#### Testing Instructions
**Atomic sites**
* Go to themes with an atomic site selected. Ex: `/themes/:site`
* Search for a WP.org theme. Ex: `astral`
* Click on the options of the given theme
* Verify the options `Activate` and `Try & Customize` MUST be present

**non-atomic sites**
* Go to themes with an atomic site selected. Ex: `/themes/:site`
* Search for a WP.org theme. Ex: `astral`
* Click on the options of the given theme
* Verify the options `Activate` and `Try & Customize` must NOT be present

| Atomic  | Non-atomic |
| ------------- | ------------- |
|Screen Shot 2022-11-18 at 16 53 08|Screen Shot 2022-11-18 at 16 51 56|

#### Pre-merge Checklist
**Non-atomic sites**

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #

### CODE REVIEW

**Comment by zdenys** [comment]
Just cross-linking to where it was first reported, for posterity: pdt2If-uX-p2#comment-291

**Review by gavande1 — APPROVED** [review]
LGTM!

**Comment by daledupreez** [comment]
I _think_ this should be OK, because the `isWpcomTheme()` checks for all `wpcom` themes, but I would love for a sanity check from someone on @Automattic/serenity with slightly more history on the themes code.

  As a related question, how are we handling themes that exist in both source systems? I believe some recent examples of this include Pixl ([.org](https://wordpress.org/themes/pixl/), [.com](https://wordpress.com/theme/pixl)) and Rainfall ([.org](https://wordpress.org/themes/rainfall/), [.com](https://wordpress.com/theme/rainfall)).

**Comment by ebinnion** [comment]
Is there any reason why we haven't shipped this PR yet? cc @cpapazoglou

**Comment by WBerredo** [comment]
@ebinnion we have been waiting for a review from @Automattic/serenity if this code matches the current state and the plans for the future on the themes area. 

Reference: p1669307988134389-slack-C029JEQRVRT

**Comment by markbiek** [comment]
Some additional Serenity discussion happening in p1670258732947949-slack-C0Q664T29

**Review by sixhours — APPROVED** [review]
Working for me!

**Review by markbiek — APPROVED** [review]
Works as expected and confirmed with the rest of @Automattic/serenity that we're good to approve and ship.

**Pull request merged**
